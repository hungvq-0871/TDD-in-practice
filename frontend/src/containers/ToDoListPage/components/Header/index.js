import React from 'react';
import PropTypes from 'prop-types';

import Button from 'components/Button';
import BtnGroup from './components/BtnGroup';
import HeaderStyle from './index.style';

import theme from 'styles/theme';

function Header({
  onSearchClick,
  onExportClick,
  onShareClick,
  onCreateNewTask
}) {
  return (
    <HeaderStyle>
      <h3>My Tasks</h3>
      <div className="right">
        <BtnGroup
          onSearchClick={onSearchClick}
          onExportClick={onExportClick}
          onShareClick={onShareClick}
        />
        <Button
          color={theme.white}
          bgColor={theme.blue2}
          bdColor={theme.blue2}
          onClick={onCreateNewTask}
        >
          New Task
      </Button>
      </div>
    </HeaderStyle>
  )
}

Header.propTypes = {
  onSearchClick: PropTypes.func.isRequired,
  onExportClick: PropTypes.func.isRequired,
  onShareClick: PropTypes.func.isRequired,
  onCreateNewTask: PropTypes.func.isRequired
}

export default Header;
