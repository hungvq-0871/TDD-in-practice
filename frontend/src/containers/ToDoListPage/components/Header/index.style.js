import styled from 'styled-components'

const HeaderStyle = styled.div`
  display: flex;
  justify-content: space-between;
`;

export default HeaderStyle;