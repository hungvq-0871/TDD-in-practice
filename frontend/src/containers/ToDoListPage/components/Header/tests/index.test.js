import React from 'react';
import { render } from '@testing-library/react';

import Header from '../index';

describe('test how "Header" is rendered', () => {
  test(`it should match it's snapshot`, () => {
    const { container } = render(<Header
      onSearchClick={() => { }}
      onExportClick={() => { }}
      onShareClick={() => { }}
      onCreateNewTask={() => { }}
    />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
