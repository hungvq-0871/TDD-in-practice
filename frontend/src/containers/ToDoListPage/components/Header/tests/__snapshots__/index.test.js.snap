// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test how "Header" is rendered it should match it's snapshot 1`] = `
<div
  class="sc-htpNat hqmgw"
>
  <h3>
    My Tasks
  </h3>
  <div
    class="right"
  >
    <div
      class="sc-bwzfXH SfLdw btn-group"
      role="group"
    >
      <button
        class="sc-bdVaJa fiYuhs btn search-btn"
      >
        Search
      </button>
      <button
        class="sc-bdVaJa fiYuhs btn export-btn"
      >
        Export
      </button>
      <button
        class="sc-bdVaJa fiYuhs btn share-btn"
      >
        Share
      </button>
    </div>
    <button
      class="sc-bdVaJa iHgvgu btn undefined"
    >
      New Task
    </button>
  </div>
</div>
`;
